sub-packages/bionemo-geneformer/src/bionemo/geneformer/data/singlecell/dataset.py

```diff
@@ -105,110 +105,30 @@ def __init__(  # noqa: D107
-                count += self.metadata[k]["shape"][0]
-            self.dataset_ccum[0] = -1
-        else:
-            # We can store a single feature_id vector for all datasets, and do not need to store the full metadata array.
-            logging.warning(
-                "Feature ids are the same across datasets. This is good, using the same feature_ids for all datasets."
-            )
-            self.feature_ids = feature_ids
-            self.metadata = None
+        self.bypass_tokenizer_vocab = bypass_tokenizer_vocab
 
     def __len__(self):  # noqa: D105
-        return self.num_samples
-
-    def metadata_lookup(self, idx) -> Dict[str, np.ndarray]:
-        """Go from a cell idx to the file-level metadata associated with that cell."""
-        did = sum(~(self.dataset_ccum > idx)) - 1
-        metadata = self.metadata[self.dataset_map[did]]
-        return metadata
-
-    def lookup_cell_by_idx(self, idx) -> Tuple[np.ndarray, np.ndarray, np.ndarray]:  # noqa: D102
-        ptr = slice(int(self.gene_data_ptr[idx]), int(self.gene_data_ptr[idx + 1]))
-        # col idxs poin to offsets in the original sparse metadata, this is for looking up metadata eg gene names
-        col_idxs = np.asarray(self.gene_data_indices[ptr]).astype(int)  # keyed by ptr
-        if self.assert_increasing_columns and len(col_idxs) > 1:
-            is_increasing = np.diff(col_idxs) > 0
-            if not np.all(is_increasing):
-                raise ValueError(f"Column indices are not increasing for {np.sum(~is_increasing)} pairs of genes")
-        gene_data = np.asarray(self.gene_data[ptr]).astype(int)  # keyed by ptr
-        # Get feature_ids for this particular cell. Eitehr lookup by index if we need to, or if we already verified that
-        #  metadata is not needed because feature_ids are the same for every file, then we can just use the single feature_ids
-        #  vector instead.
-        feature_ids: np.ndarray = (
-            self.feature_ids if self.metadata is None else self.metadata_lookup(idx)["feature_ids"]
-        )
-        return gene_data, col_idxs, feature_ids
+        return len(self.scdl)
 
     def __getitem__(self, index: EpochIndex) -> types.BertSample:
         """Performs a lookup and the required transformation for the model."""
         rng = np.random.default_rng([self._seed, index.epoch, index.idx])
-        gene_data, col_idxs, feature_ids = self.lookup_cell_by_idx(index.idx)
+        values, feature_ids = self.scdl.get_row(index.idx, return_features=True, feature_vars=["feature_id"])
+        gene_data, col_idxs = np.array(values[0]), np.array(values[1])
+        if len(gene_data) == 0:
+            raise ValueError(
+                "SingleCellMemap data provided is invalid; the gene expression data parsed for the specified index is empty."
+            )
         return process_item(
             gene_data,
             col_idxs,
-            feature_ids,
+            feature_ids[0],
```

Comment:
Why are we only returning the first value here?

Reply:
I hope it's because feature_ids is a length 1 list always? If so I would add an assert len(feature_ids) == 1 or something before pulling out the 0th element. A comment about the shapes etc would be helpful.